packages/trader/src/App/Components/Elements/VideoPlayer/volume-control.tsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import classNames from 'classnames';
+import { CSSTransition } from 'react-transition-group';
+import { Icon } from '@deriv/components';
+
+type TVolumeControl = {
+    onVolumeChange: (new_value: number) => void;
+    volume?: number;
+    is_mobile?: boolean;
+    is_muted?: boolean;
+    toggleMute: (new_value: boolean) => void;
+};
+
+const VolumeControl = ({ onVolumeChange, volume, is_mobile, is_muted, toggleMute }: TVolumeControl) => {
+    const [is_animated, setIsAnimated] = React.useState(true);
+    const [show_volume, setShowVolume] = React.useState(false);
+    const [shift_Y, setShiftY] = React.useState<number>(0);
+
+    const volume_bar_filled_ref = React.useRef<HTMLDivElement>(null);
+    const volume_bar_ref = React.useRef<HTMLDivElement>(null);
+    const volume_dot_ref = React.useRef<HTMLSpanElement>(null);
+    const is_dragging = React.useRef<boolean>(false);
```

Comment:
```suggestion
    const is_dragging = React.useRef(false);
```
does TS infer the type successfully here? :)